terraform/infrastructure/azure/main.tf

```diff
@@ -232,6 +241,28 @@ resource "azurerm_network_security_group" "security_group" {
   }
 }
 
+resource "azurerm_network_security_rule" "nsg_rule" {
+  for_each = {
+    for o in concat(
+      local.ports,
+      [{ name = "nodeports", port = local.ports_node_range, priority = 200 }]
```

Comment:
Why do we need to expose node ports via LB? Do we need to expose the `ports_node_range` for UDP, too?

Reply:
I don't know if I fully get your question. The intent of the code is to allow LBs to route traffic to node on the nodeports. The default case when using an LB is that the node opens a nodeport and the LB routes to the nodeip:nodeport. Good catch with the UDP though.

Reply:
I read this block as adding a firewall rule for the entire NodePort range by default, but I expected individual exceptions for load balancers that are actually deployed.

Reply:
Good point. Azure has those LB to VM allow rule that exists by default and the CCM creates the needed "port to LB public IP" rule. Removed the nodeports from here.